### DISCUSSION TITLE
Migration from snap nextcloud (mysql) to aio

### DISCUSSION THREAD

**Discussion by jannessm** [nextcloud/all-in-one#5564] [Wiki]
# Documentation enhancement
In the migration description, it is stated that a migration from snap to aio is not possible. After a lot of hours, I found a way to migrate. Here is my procedure that could be added to your documentation.

# Migration from snap to aio
1. use the snap export functionality to dump the mysql database
```bash
nextcloud.export -a
```
2. download nextcloud as a zip and setup your env to be able to execute the occ command `https://nextcloud.com/install/#instructions-server`
3. If the occ db:convert-type cannot be executed, follow https://github.com/nextcloud/server/issues/45257#issuecomment-2363167396 and modify the scripts accordingly. You might disable `https://github.com/nextcloud/server/blob/master/core/Command/Db/ConvertType.php#L161`
4. Follow the instructions in the migration file.

Hope this helps anyone

**Comment by szaimen** [comment]
Hi, I suppose easier would be to set up a temporary docker instance in between and migrate from snap to docker and then to aio.
See https://github.com/nextcloud/docker/blob/master/.examples/docker-compose/insecure/mariadb/apache/compose.yaml
